src/app/components/query-explainer/query-explainer.component.ts

```diff
@@ -0,0 +1,129 @@
+import { BaseComponent } from "src/app/common/base.component";
+import { MaybeNull, MaybeUndefined } from "src/app/common/app.types";
+import ProjectLinks from "src/app/project-links";
+import { catchError, forkJoin, Observable, of, switchMap, tap } from "rxjs";
+import { QueryExplainerResponse, VerifyQueryResponse } from "./query-explainer.types";
+import { HttpErrorResponse } from "@angular/common/http";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import { DataRow, DataSchemaField } from "src/app/interface/dataset.interface";
+import { extractSchemaFieldsFromData } from "src/app/common/table.helper";
+import { BlockService } from "src/app/dataset-block/metadata-block/block.service";
+
+@Component({
+    selector: "app-query-explainer",
+    templateUrl: "./query-explainer.component.html",
+    styleUrls: ["./query-explainer.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueryExplainerComponent extends BaseComponent implements OnInit {
+    private clipboard = inject(Clipboard);
+    private queryExplainerService = inject(QueryExplainerService);
+    private cdr = inject(ChangeDetectorRef);
+    private blockService = inject(BlockService);
+
+    private sqlQuery: MaybeNull<string>;
+    public readonly DATE_FORMAT = AppValues.DISPLAY_FLOW_DATE_FORMAT;
+
+    public sqlQueryExplainer: QueryExplainerResponse;
+    public sqlQueryVerify$: Observable<VerifyQueryResponse>;
+    public verifyResponse: VerifyQueryResponse;
+    public blockHashSystemTimes: string[] = [];
+
+    ngOnInit(): void {
+        this.sqlQuery = this.extractSqlQueryFromRoute();
+
+        if (this.sqlQuery) {
+            const blockHashObservables$: Observable<string>[] = [];
+            this.queryExplainerService
+                .proccessQuery(this.sqlQuery)
+                .pipe(
+                    tap((response) => {
+                        this.sqlQueryExplainer = response;
+                        response.input.datasets
+                            .map((dataset) => ({ datasetId: dataset.id, blockHash: dataset.blockHash }))
+                            .forEach(({ datasetId, blockHash }) => {
+                                blockHashObservables$.push(
+                                    this.blockService.requestSystemTimeBlockByHash(datasetId, blockHash),
+                                );
+                            });
+                    }),
+                    switchMap((response: QueryExplainerResponse) => {
+                        return this.queryExplainerService.verifyQuery(response);
+                    }),
+                    catchError((e: HttpErrorResponse) => {
+                        return of(e.error as VerifyQueryResponse);
+                    }),
+                    tap((data) => {
+                        this.verifyResponse = data;
+                    }),
+                    switchMap(() => forkJoin(blockHashObservables$)),
```

Comment:
`blockHashObservables$` as a local variable... this is whatever, but not functional reactive programming.